File: frontend/src/components/services/ProductService.js
import axios from 'axios';

// const Base_URL = 'http://localhost:8080/product';
const Base_URL = 'https://fakestoreapi.com/products';

export class ProductService{
    static getAllProducts(){
        return axios.get(`${Base_URL}`);
    }
    static getProductById(id){
        return axios.get(`${Base_URL}/${id}`);
    }
    
}

File: microservices/product-service/src/main/java/com/grocery/productservice/service/ProductService.java
package com.grocery.productservice.service;

import com.grocery.productservice.entity.Product;

import java.util.List;

public interface ProductService {
    List<Product> getAllProducts();
    Product getProductById(Long id);
    Product createProduct(Product product);
    Product updateProduct(Long id, Product product);
    void deleteProduct(Long id);
    List<Product> searchProductsByName(String name);
    List<Product> searchProductsByCategoryId(Long categoryId);
}
